### ISSUE TITLE
has disclaimed fix

### ISSUE THREAD

**Issue by shrugs** [urbit/bridge#208]
the has disclaimed update isn't actually functional. with the following code

```js
  const goBack = useCallback(() => {
    setHasDisclaimed(true);
    pop();
  }, [pop, setHasDisclaimed]);
```

what actually happens is:

1. `setState` in `useHasDisclaimed` fires synchronously (great 👍 )
2. `pop()` unmounts the current component (great 👍 )
3. react cancels the `useEffect` from `useHasDisclaimed` before the new value is written to local storage (bad 🙅‍♂)

solution 1:

```js 
import timeout from 'lib/timeout'

const tick = () => timeout(0)

...

const goBack = useCallback(async () => {
    setHasDisclaimed(true);
    await tick();
    pop();
  }, [pop, setHasDisclaimed]);
```

which lets the effects be flushed before the pop unmounts them

solution 2:
a synchronous setState that wraps the normal `setState`

---

after writing all this out, I'm going to do option 2 cause it's the least leaky and we're talking nanoseconds at worse, afaik

**Comment by Fang-** [comment]
Good god, what a minefield. ):

Nice catch, agreed that solution 2 seems better!

**Comment by shrugs** [comment]
closed in #209
